### PR TITLE
fixes setting paramName only when it is not null

### PR DESCRIPTION
**Pull request by ankurpathak**
Fixes: gh-6223

### CODE REVIEW

File: core/src/test/java/org/springframework/security/access/expression/method/MethodSecurityEvaluationContextTests.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+			extends MethodSecurityEvaluationContext {
+
+		public NotNullVariableMethodSecurityEvaluationContext(Authentication auth, MethodInvocation mi,
+				ParameterNameDiscoverer parameterNameDiscoverer) {
+			super(auth, mi, parameterNameDiscoverer);
+		}
+
+		@Override
+		public void setVariable(String name, @Nullable Object value) {
+			if (name == null)
```
Comment (rwinch) [inline core/src/test/java/org/springframework/security/access/expression/method/MethodSecurityEvaluationContextTests.java]: Please add `{` to follow Spring Security's code conventions

File: core/src/main/java/org/springframework/security/access/expression/method/MethodSecurityEvaluationContext.java [hunk 2]
```diff
@@ -112,7 +112,9 @@ private void addArgumentsAsVariables() {
 		}
 
 		for (int i = 0; i < args.length; i++) {
-			super.setVariable(paramNames[i], args[i]);
+			if (paramNames[i] != null) {
+				super.setVariable(paramNames[i], args[i]);
```
Comment (rwinch) [inline core/src/main/java/org/springframework/security/access/expression/method/MethodSecurityEvaluationContext.java]: For the test below to work, you need to remove `super.`

File: core/src/test/java/org/springframework/security/access/expression/method/MethodSecurityEvaluationContextTests.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	private ParameterNameDiscoverer paramNameDiscoverer;
+	@Mock
+	private Authentication authentication;
+	@Mock
+	private MethodInvocation methodInvocation;
+	@Mock
+	private AuthenticationTrustResolver trustResolver;
+
+	@Test
+	public void setVariableTest() {
```
Comment (rwinch) [inline core/src/test/java/org/springframework/security/access/expression/method/MethodSecurityEvaluationContextTests.java]: please rename to `lookupVariableWhenParameterNameNullThenNotSet`

File: core/src/test/java/org/springframework/security/access/expression/method/MethodSecurityEvaluationContextTests.java (hunk 1 above)
Comment (ankurpathak) [reply core/src/test/java/org/springframework/security/access/expression/method/MethodSecurityEvaluationContextTests.java]: Added { to if part.

Comment (ankurpathak) [reply core/src/test/java/org/springframework/security/access/expression/method/MethodSecurityEvaluationContextTests.java]: Do we need to add it for else part as well??

File: core/src/main/java/org/springframework/security/access/expression/method/MethodSecurityEvaluationContext.java (hunk 2 above)
Comment (ankurpathak) [reply core/src/main/java/org/springframework/security/access/expression/method/MethodSecurityEvaluationContext.java]: Removed super.

File: core/src/test/java/org/springframework/security/access/expression/method/MethodSecurityEvaluationContextTests.java (hunk 3 above)
Comment (ankurpathak) [reply core/src/test/java/org/springframework/security/access/expression/method/MethodSecurityEvaluationContextTests.java]: renamed test to lookupVariableWhenParameterNameNullThenNotSet

File: core/src/test/java/org/springframework/security/access/expression/method/MethodSecurityEvaluationContextTests.java (hunk 1 above)
Comment (rwinch) [reply core/src/test/java/org/springframework/security/access/expression/method/MethodSecurityEvaluationContextTests.java]: Yes. We do not use implied `{` for one line statements

**Comment by ankurpathak** [comment]
changed commit message to reflect Fixes: gh-6223.

**Comment by rwinch** [comment]
Thanks @ankurpathak! This is now merged into master

**Pull request merged**